doc/usage/restructuredtext/domains.rst

```diff
@@ -1225,6 +1225,8 @@ There is a set of directives allowing documenting command-line programs:
    The directive will create cross-reference targets for the given options,
    referenceable by :rst:role:`option` (in the example case, you'd use something
    like ``:option:`dest_dir```, ``:option:`-m```, or ``:option:`--module```).
+   When used within the :rst:dir:`program` directive, however, it is mandatory
```

Comment:
If you leave out the program name, Sphinx fails as I showed in #6221.

In order for Sphinx not to give warnings and be able to resolve references it seems, one *must* specify the program name (hence mandatory).

(I see, you pointed out a [special case](https://github.com/sphinx-doc/sphinx/issues/6221#issuecomment-478344019), where the reference in fact does resolve without program name when it's used *after* the program directive. Note, that this only works within the same file. Also, I would deem this behavior counter-intuitive and undesirable. Also I fail to find it documented anywhere.)

In case I misunderstood, could you please provide an example how it is optional when used with the program directive?

Reply:
What I meant is a "special case" you mentioned. And I agree with the behavior is not documented. So it would be better to describe the behavior of option role including how program directive works.